python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -121,6 +122,13 @@ def __init__(
             self._default_load_type = default_load_type
 
     def handle_output(self, context: OutputContext, obj: object) -> None:
+        # Nones cannot be stored by DB I/O managers. If the output type is set to None/Nothing,
+        # the handle_output will not be called
+        if obj is None:
+            raise DagsterInvariantViolationError(
```

Comment:
confirm this is the right exception to raise here

Reply:
ya this is general "problem could not be detected til runtime", not sure if theres a more specific appropriate exception